skyportal/handlers/api/gcn.py

```diff
@@ -1033,9 +1056,98 @@ def add_tiles(localization_id):
         session.add(localization)
         session.add_all(tiles)
         session.commit()
-        log(f"Generated tiles for localization {localization_id}")
+
+        config_gcn_observation_plans_all = [
+            observation_plan for observation_plan in cfg["gcn.observation_plans"]
+        ]
+        config_gcn_observation_plans = []
+        for config_gcn_observation_plan in config_gcn_observation_plans_all:
+            allocation = session.scalars(
+                Allocation.select(user).where(
+                    Allocation.proposal_id
+                    == config_gcn_observation_plan["allocation-proposal_id"]
+                )
+            ).first()
+            if allocation is not None:
+                config_gcn_observation_plan["allocation_id"] = allocation.id
+                config_gcn_observation_plan["survey_efficiencies"] = []
+                config_gcn_observation_plans.append(config_gcn_observation_plan)
+
+        default_observation_plans = (
+            (
+                session.scalars(
+                    DefaultObservationPlanRequest.select(
+                        user,
+                        options=[
+                            joinedload(
+                                DefaultObservationPlanRequest.default_survey_efficiencies
+                            )
+                        ],
+                    )
+                )
+            )
+            .unique()
+            .all()
+        )
+        gcn_observation_plans = []
+        for plan in default_observation_plans:
+            allocation = session.scalars(
+                Allocation.select(user).where(Allocation.id == plan.allocation_id)
+            ).first()
+
+            gcn_observation_plan = {}
```

Comment:
Not very important, I'll let you decide if you change it or not, but here you could just create the object/dict directly, rather than instantiate it and set each key/value one by one.

Reply:
fixed.